[TTAHUB-2546] Small changes to TR goal deletion - missed from previous PR

nvms opened the pull request:
## Description of change

This was supposed to be in the last PR, but I forgot to push it. It's a pretty small change and the logic should be simple enough to verify by just looking at it, but please go ahead and test using the steps below if you'd like.

The change:

Don't delete the goal if it is currently in use by another session. In other words, only delete the goal if the session removing this recipient is the ONLY session using this recipient/goal pair.

## How to test

Useful query:

```sql
SELECT e.id AS "ERPG id",
       erp.data->>'goal' as "EventReportPilot.data->>'goal'",
       g.id AS "Goal.id",
       g.name AS "Goal.name",
       g."grantId" AS "Goal.grantId",
       g.status AS "Goal.status",
       g."previousStatus" AS "Goal.previousStatus",
       s.data->>'status' AS "SessionReportPilot.data->>'status'"
FROM "EventReportPilotGoals" as e
LEFT JOIN "EventReportPilots" as erp ON e."eventId" = erp.id
LEFT JOIN "Goals" as g ON e."goalId" = g.id
LEFT JOIN "SessionReportPilots" as s ON e."sessionId" = s.id
WHERE e."goalId" IS NOT NULL;
```

- Session1: add two recipients (they should both get a unique goal created for them):

```bash
 ERPG id | EventReportPilot.data->>'goal' | Goal.id | Goal.name | Goal.grantId | Goal.status | Goal.previousStatus | SessionReportPilot.data->>'status' 
---------+--------------------------------+---------+-----------+--------------+-------------+---------------------+------------------------------------
       1 | asdf                           |       5 | asdf      |           10 | Draft       |                     | In progress
       2 | asdf                           |       6 | asdf      |           18 | Draft       |                     | In progress

```
- Session2: add both of the same recipients, no new goals are created and the result of the query should look exactly the same as above.
- Session2: remove ONE of the recipients, and re-run the query to verify no goals have been deleted because they are in use by that recipient on session1: the result of the query should look exactly the same as above.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2546


## Checklists

### Every PR

<!-- Add details to each completed item -->
- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

thewatermethod commented:
Using seeded training report, otherwise fresh data

- Create a session for a training event. Added three recipients
  - SELECT * FROM "EventReportPilotGoals
  - Looks good. Session ID: 1, 3 goal IDs, 3 grant IDs
- Add a second session for that training event. Added same three recipients
  - SELECT * FROM "EventReportPilotGoals
  - Session ID: 1, 3 goal IDs, 3 grant IDs. Nothing referencing session 2. Correct?
- Removed a recipient from session 2.
  - Same query, same result. 
- Remove same recipient from session 1   
  - Same query, same result? 3 columns, session 1, all three recipients

nvms commented:
> Using seeded training report, otherwise fresh data
> 
> * Create a session for a training event. Added three recipients
>   
>   * SELECT * FROM "EventReportPilotGoals
>   * Looks good. Session ID: 1, 3 goal IDs, 3 grant IDs
> * Add a second session for that training event. Added same three recipients
>   
>   * SELECT * FROM "EventReportPilotGoals
>   * Session ID: 1, 3 goal IDs, 3 grant IDs. Nothing referencing session 2. Correct?

Yep! Creating a new one at that point would just be a duplicate for the same recipient.

> * Removed a recipient from session 2.
>   
>   * Same query, same result.
> * Remove same recipient from session 1
>   
>   * Same query, same result? 3 columns, session 1, all three recipients

This is the correct behavior. All three recipients have their 3 goals.

The deletion case happens when the recipient isn't used any more on any session.

To keep going, you could go back to session 1 and remove one of the recipients. You should see a goal get destroyed.

thewatermethod commented:
> Edit: Oh wait I misread - you removed the recip from session 1 and a goal wasn't destroyed? I just tried reproducing this and can't, hmm 

I will run through again

thewatermethod commented:
@nvms I restarted my environment and re-ran through the steps and it worked the second time, I guess it just didn't pick up the changes to the hooks or something the first time
